tests/spark/sdk/cluster/test_cluster.py

```diff
@@ -0,0 +1,143 @@
+import subprocess
+from datetime import datetime
+
+import azure.batch.models as batch_models
+from azure.batch.models import BatchErrorException
+
+import aztk.spark
+import pytest
+from aztk.error import AztkError
+from cli import config
+
+
+dt = datetime.now()
+time = dt.microsecond
+cluster_id = "test{}".format(time)
+
+# load secrets
+# note: this assumes secrets are set up in .aztk/secrets
+spark_client = aztk.spark.Client(config.load_aztk_screts())
+
+
+def test_create_cluster():
+    # TODO: make Cluster Configuration more robust, test each value
+    cluster_configuration = aztk.spark.models.ClusterConfiguration(
+        cluster_id=cluster_id,
+        vm_count=2,
+        vm_low_pri_count=0,
+        vm_size="standard_f2",
+        subnet_id=None,
+        custom_scripts=None,
+        file_shares=None,
+        docker_repo=None,
+        spark_configuration=None
+    )
+    try:
+        cluster = spark_client.create_cluster(cluster_configuration, wait=True)
+    except (AztkError, BatchErrorException) as e:
+        assert False
+
+    assert cluster.pool is not None
+    assert cluster.nodes is not None
+    assert cluster.id == cluster_id
+    assert cluster.vm_size == "standard_f2"
+    assert cluster.current_dedicated_nodes == 2
+    assert cluster.gpu_enabled is False
+    assert cluster.master_node_id is not None
+    assert cluster.current_low_pri_nodes == 0
+
+def test_get_cluster():
+    try:
+        cluster = spark_client.get_cluster(cluster_id=cluster_id)
+    except (AztkError, BatchErrorException) as e:
+        assert False
+
+    assert cluster.pool is not None
+    assert cluster.nodes is not None
+    assert cluster.id == cluster_id
+    assert cluster.vm_size == "standard_f2"
+    assert cluster.current_dedicated_nodes == 2
+    assert cluster.gpu_enabled is False
+    assert cluster.master_node_id is not None
+    assert cluster.current_low_pri_nodes == 0
+
+def test_list_clusters():
+    try:
+        clusters = spark_client.list_clusters()
+    except (AztkError, BatchErrorException) as e:
+        assert False
+    assert cluster_id in [cluster.id for cluster in clusters]
```

Comment:
should we be testing more in this method? Like the number of clusters we expect to see?

Reply:
Since this can execute in any batch account, we couldn't be sure how many clusters there should be. Even if we get the count of clusters before we create one as part of these tests, we can't be sure that it will remain.